### PR TITLE
Add command to sign a withdrawal credentials update message using a validator keystore

### PR DESCRIPTION
**Pull request by valefar-on-discord**
There is a cohort of users who have lost or are missing their mnemonic and are unable to update their withdrawal credentials. As of now, these funds are locked permanently and indefinitely. We have a potential solution that can allow these users to regain their funds though we have no approval or proper implementation. One of the necessary components of this solution is a signature from the validator keystore that can be used as a proof of ownership of the validator.  This adds the ability for a user to generate said signature.

The users runs `generate-bls-to-execution-change-keystore` with the provided keystore, password, validator index, and designed execution address to update their withdrawal credentials with.  This command outputs and signature json file where the message will contain the provided execution address and validator index.

### CODE REVIEW

**Comment by valefar-on-discord** [comment]
I'm thinking about changing the logic to be a more generic "sign message with validator keystore"...

File: ethstaker_deposit/utils/constants.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
 DOMAIN_DEPOSIT = bytes.fromhex('03000000')
 DOMAIN_VOLUNTARY_EXIT = bytes.fromhex('04000000')
 DOMAIN_BLS_TO_EXECUTION_CHANGE = bytes.fromhex('0A000000')
+DOMAIN_BLS_TO_EXECUTION_CHANGE_KEYSTORE = bytes.fromhex('0B000000')
```
Comment (valefar-on-discord) [inline ethstaker_deposit/utils/constants.py:14]: This needs to be changed: https://github.com/prysmaticlabs/prysm/blob/7c213ce161cb929659c9cd96c7bfa08bcd8ef820/config/params/mainnet_config.go#L174C2-L174C21

**Comment by remyroy** [comment]
It would be nice if you could link the related proposals if they exist in a draft or a working document somewhere else.

**Comment by valefar-on-discord** [comment]
> It would be nice if you could link the related proposals if they exist in a draft or a working document somewhere else.

The only location for the proposal right now is https://github.com/eth-educators/update-credentials-without-mnemonic

If we are in agreement, my plan would be to get this landed (when ready) and create a new release. I'll then update the other repo with the new instructions and then we can broadcast to determine interest.

**Comment by yorickdowne** [comment]
Do you want the logic as-is in here, or do you prefer the more generic "sign message" logic?

**Comment by valefar-on-discord** [comment]
After some thought I'd prefer to keep it as is.

The generic signing came as an idea with the Stakers Union project where you have to turn off your validator to prove ownership. My thought was signing the Stakers Union's Oath with the validator keystore would do just as well.

But I am also of the opinion that any form of generic signing is dangerous. This needs to be a highly secure project based on the operations. I would rather build out a separate command for that use case if the need arises.

**Comment by yorickdowne** [comment]
OK, I agree. I am interested in seeing this in. Even if we end up backing it out again, if there’s just not appetite to actually do this, from the Ethereum devs.

Let’s see what Remy says when he’s back next week.

**Comment by remyroy** [comment]
I wish there was more efforts into bringing this as an official EIP and gathering comments and reviews from peers in the ecosystem. I'm happy to review this PR and include it in the project but I fear it will not be used if the principles and the specs behind this feature isn't accepted at the protocol level.

**Comment by yorickdowne** [comment]
We’d include this to gauge interest and see how many people are impacted. That’s how we get the protocol on board. Chicken and egg.

At some point it’ll be removed again.

**Review by remyroy — CHANGES_REQUESTED** [review]
One small change I would make to the documentation and everything else seems good.

File: docs/src/generate_bls_to_execution_change_keystore.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# generate-bls-to-execution-change-keystore
+
+<div class="warning">
+This command is associated with the a proposed solution to update withdrawal credentials for those who are missing their mnemonic. At this point this has not been approved or implemented and there is no guarantee credentials will be modified in the future.
```
[the remark this answers is not in the document]
Comment (valefar-on-discord) [reply docs/src/generate_bls_to_execution_change_keystore.md:4]: Added a reference

**Pull request merged**